### PR TITLE
chore(stdlib)!: Update Buffer/Bytes to use new integer types

### LINKED ISSUE #1701: Update bytes/buffer to return (u)int8 and (u)int16 where appropriate
In #1669, the various small integer types were added to the language. We need to update the bytes and buffer stdlib to return those types where appropriate—e.g. `getInt8`.

### PR DESCRIPTION
**Pull request by alex-snezhko**
Closes #1701, also adds functions for `Uint32/Uint64`

Also fixes some short int parsing behavior that I felt was simple enough to also include in this PR (allowing `0xffs` for instance)

### CODE REVIEW

**Comment by phated** [comment]
Looks like the compiler changes need to be formatted

File: compiler/src/utils/literals.re [hunk 1]
```diff
@@ -71,29 +71,57 @@ let int8_max = 127l;
...
+    String.length(s) > 2 && List.mem(String.sub(s, 0, 2), ["0x", "0b"]);
   switch (Int32.of_string_opt(s)) {
   | None => None
   | Some(n) =>
-    if (n < int8_min || n > int8_max) {
+    let (&) = Int32.logand;
+    let (>>) = Int32.shift_right;
+    let (<<) = Int32.shift_left;
+    if (n > int8_max) {
+      // Trick to get something like 0xFF to be legal
```
Comment (ospencer) [inline compiler/src/utils/literals.re:83]:
```suggestion
      // Trick to allow setting the sign bit in representations like 0xFF
```

File: compiler/src/utils/literals.re [hunk 2]
```diff
@@ -71,29 +71,57 @@ let int8_max = 127l;
...
+    String.length(s) > 2 && List.mem(String.sub(s, 0, 2), ["0x", "0b"]);
   switch (Int32.of_string_opt(s)) {
   | None => None
   | Some(n) =>
-    if (n < int16_min || n > int16_max) {
+    let (&) = Int32.logand;
+    let (>>) = Int32.shift_right;
+    let (<<) = Int32.shift_left;
+    if (n > int16_max) {
+      // Trick to get something like 0xFFFF to be legal
```
Comment (ospencer) [inline compiler/src/utils/literals.re:112]:
```suggestion
      // Trick to allow setting the sign bit in representations like 0xFFFF
```

File: compiler/src/utils/literals.re [hunk 3]
```diff
@@ -71,29 +71,57 @@ let int8_max = 127l;
 let int8_min = (-128l);
 
 let conv_int8 = s => {
+  let hex_or_bin =
```
Comment (ospencer) [inline compiler/src/utils/literals.re:74]: Shouldn't this account for octal as well?

**Comment by ospencer** [comment]
Looks like you need another compiler format.

**Review by ospencer — APPROVED** [review]
Looks great!

**Pull request merged**